source/frontend/src/features/documents/hooks/useDocumentProvider.ts

```diff
@@ -208,6 +211,29 @@ export const useDocumentProvider = () => {
       }, []),
     });
 
+  // Provides functionality for stream a document file
+  const { execute: streamDocumentFile, loading: streamDocumentFileLoading } = useApiRequestWrapper<
+    (
+      mayanDocumentId: number,
+      mayanFileId: number,
+    ) => Promise<AxiosResponse<AxiosResponse<File, any>>>
+  >({
+    requestFunction: useCallback(
+      async (mayanDocumentId: number, mayanFileId: number) =>
+        await streamDocumentFileApiCall(mayanDocumentId, mayanFileId),
+      [streamDocumentFileApiCall],
+    ) as unknown as () => Promise<AxiosResponse<AxiosResponse<File, any>>>,
```

Comment:
why is the unknown necessary?

Reply:
Yeah, this is a headache, but currently I haven't found a good way to tell Typescript that when the rawResponse flag is set, the response returns the whole axios response not just the data.

Reply:
ah, yes. I saw this before, we need to make our response type more broad or add another result type on the wrapper.